src/functional/geometry/convex_hull.zig

```diff
@@ -204,3 +345,77 @@ test "incremental convex hull random elements" {
         try testing.expectEqual(turn, Turn.left);
     }
 }
+
+test "Compute MABR LinearFunction for known ConvexHull one" {
+    const allocator = std.testing.allocator;
+
+    // Define a set of points forming a simple rectangle.
+    const points = [_]DiscretePoint{
+        .{ .time = 0, .value = 0.0 },
+        .{ .time = 1, .value = 3.0 },
+        .{ .time = 2, .value = 2.0 },
+        .{ .time = 3, .value = 5.0 },
+        .{ .time = 4, .value = 4.0 },
+    };
+
+    // Initialize the convex hull and add points to it.
+    var convex_hull = try ConvexHull.init(allocator);
+    defer convex_hull.deinit();
+
+    for (points) |point| {
+        try convex_hull.addPoint(point);
+    }
+
+    // Calculate the minimum-area bounding rectangle (MABR).
+    const mabr_linear_function = try convex_hull.computeMABRLinearFunction();
+
+    // Known rectangle area and known LinearFunction y=x+1
+    try testing.expect(@abs(mabr_linear_function.slope - 1) <= 0.0001);
+    try testing.expect(@abs(mabr_linear_function.intercept - 1) <= 0.0001);
+}
+
+test "Compute MABR LinearFunction for known ConvexHull two" {
+    const allocator = std.testing.allocator;
+
+    // Define a set of points forming a simple rectangle.
+    const points = [_]DiscretePoint{
+        .{ .time = 0, .value = 0.0 },
+        .{ .time = 1, .value = -2.0 },
+        .{ .time = 2, .value = 3.0 },
+        .{ .time = 3, .value = -1.0 },
+        .{ .time = 4, .value = 2.0 },
+    };
+    // Initialize the convex hull and add points to it.
+    var convex_hull = try ConvexHull.init(allocator);
+    defer convex_hull.deinit();
+
+    for (points) |point| {
+        try convex_hull.addPoint(point);
+    }
+
+    // Calculate the minimum-area bounding rectangle (MABR).
+    const mabr_linear_function = try convex_hull.computeMABRLinearFunction();
+
+    // Known rectangle area and known LinearFunction y=x+1
+    try testing.expect(@abs(mabr_linear_function.slope - 1.5) <= 0.0001);
+    try testing.expect(@abs(mabr_linear_function.intercept + 2.75) <= 0.0001);
+}
+
+test "Compute MABR LinearFunction for random ConvexHull" {
+    const num_points: usize = 1000;
+    const allocator = testing.allocator;
+    var rnd = std.rand.DefaultPrng.init(0);
+
+    var convex_hull = try ConvexHull.init(allocator);
+    defer convex_hull.deinit();
+
+    for (0..num_points) |i| {
+        try convex_hull.addPoint(.{ .time = i, .value = rnd.random().float(f64) });
+    }
+
+    // Calculate the minimum-area bounding rectangle (MABR). If dont have way to check if the
+    // result is correct but we can check that no error has happened.
+    const mabr_linear_function = try convex_hull.computeMABRLinearFunction();
+
+    try testing.expect(mabr_linear_function.intercept < 1.0);
```

Comment:
Unclear how it is known that the intercept is below zero and why it is important to check.

Reply:
I thought to use this test to check that no errors occur with a random set of numbers. Nevertheless, since all points are between 0-1 then the intercept must be less than 1. This is because the linear function is the height bisector and goes through the midpoint of the rectangle side which has as lower and upper bounds 0 and 1.